test/hooks.js

```diff
@@ -945,3 +946,42 @@ test('onResolution hooks should be able to add extensions data', async t => {
     }
   })
 })
+
+test('onExtendSchema hooks should be triggered when extendSchema is called', async t => {
+  t.plan(1)
+
+  const app = await createTestServer(t)
+
+  app.graphql.addHook('onExtendSchema', async (schema, context) => {
+    t.pass('onExtendSchema called')
+  })
+
+  const extendedSchema = `
+    extend type Query {
+      sub(x: Int, y: Int): Int
+    }
+  `
+
+  const extendedResolvers = {
+    Query: {
+      sub: async (_, { x, y }) => x - y
+    }
+  }
+
+  await app.register(async function (app) {
+    app.graphql.extendSchema(extendedSchema)
+    app.graphql.defineResolvers(extendedResolvers)
+  })
+})
+
+test('onExtendSchema hooks should not be triggered if extendSchema is not called', async t => {
```

Comment:
what do you mean?

Reply:
try calling extend schema. this test will still pass

Reply:
actually not, I tried and I got: `expected stub to not have been called but was called once`

Reply:
do something async within the callback

Reply:
Like this?

```js
app.graphql.addHook('onExtendSchema', async (schema, context) => {
  return Promise.resolve().then(onExtendSchemaFn)
})
```

Reply:
yep

Reply:
> yep

also with this callback the test fails

Reply:
you need to call extendSchema :)

Reply:
this test passes and it shouldn't. by having a sync operation in the hook implementation, you're making the test pass artificially. rather, use an async operation within the hook to mimic a realistic situation, and change the test so that it fails if extendSchema is called

```js
  const onExtendSchemaFn = sinon.stub()

  const app = await createTestServer(t)

  app.graphql.addHook('onExtendSchema', async (schema, context) => {
    return Promise.resolve().then(onExtendSchemaFn)
  })

  app.graphql.extendSchema(`type Whatever { foo: String }`)

  sinon.assert.notCalled(onExtendSchemaFn)
```

Reply:
but this test pass because, since you are using the `onExtendSchema`, the extendSchema is returning a promise and so you need to await for it. If you use `await app.graphql.extendSchema('type Whatever { foo: String }')` the test fails as expected.

Reply:
okay fair enough, what I was trying to say is that if the production code is not implemented correctly and for whatever reason extendSchema IS called, this test will still succeed IF the hook is implemented in a realistic way, that is via a function that does something asynchronous. But ok, it's unlikely to happen so we can keep it as it is